astropy/io/misc/hdf5.py

```diff
@@ -266,11 +279,18 @@ def write_table_hdf5(table, output, path=None, compression=False,
     if serialize_meta:
         header_yaml = meta.get_yaml_from_table(table)
 
-        try:
-            dset.attrs[META_KEY] = [h.encode('utf8') for h in header_yaml]
-        except Exception as e:
-            warnings.warn("Attributes could not be written to the output HDF5 "
-                          "file: {0}".format(e))
+        header_encoded = [h.encode('utf-8') for h in header_yaml]
+        if compatibility_mode:
```

Comment:
I added the warning, but I'm not sure I understand the part about the decorator. I couldn't find a way to use any of the decorators in `decorators.py`, and in particular the one about deprecated keywords, as I'm not substituting a keyword but adding one. Could you explain in more detail?